modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/events/LoginPostAction.java

```diff
@@ -104,33 +111,57 @@ private void _addDefaultAccountRoles(HttpServletRequest httpServletRequest)
 		}
 	}
 
-	private void _updateGuestCommerceOrder(
-			String commerceOrderUuid, long commerceChannelGroupId,
-			HttpServletRequest httpServletRequest)
-		throws Exception {
+	private AccountEntry _getAccountEntry(
+		int commerceSiteType, HttpServletRequest httpServletRequest,
+		long userId) {
 
-		CommerceOrder commerceOrder = null;
+		AccountEntry accountEntry = null;
 
 		try {
-			commerceOrder =
-				_commerceOrderLocalService.getCommerceOrderByUuidAndGroupId(
-					commerceOrderUuid, commerceChannelGroupId);
-		}
-		catch (Exception exception) {
-			if (_log.isDebugEnabled()) {
-				_log.debug(exception);
+			if (commerceSiteType == CommerceChannelConstants.SITE_TYPE_B2B) {
+				accountEntry = _getBusinessAccountEntry(userId);
 			}
+			else if (commerceSiteType ==
+						CommerceChannelConstants.SITE_TYPE_B2X) {
 
-			return;
+				accountEntry = _getBusinessAccountEntry(userId);
+
+				if (accountEntry == null) {
+					accountEntry = _getPersonAccountEntry(
+						userId, httpServletRequest);
+				}
+			}
+			else {
+				accountEntry = _getPersonAccountEntry(
+					userId, httpServletRequest);
+			}
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException);
 		}
 
-		if (commerceOrder.getCommerceAccountId() !=
-				AccountConstants.ACCOUNT_ENTRY_ID_GUEST) {
+		return accountEntry;
+	}
 
-			return;
+	private AccountEntry _getBusinessAccountEntry(long userId)
+		throws PortalException {
+
+		List<AccountEntry> userAccountEntries =
+			_accountEntryLocalService.getUserAccountEntries(
+				userId, null, null,
+				new String[] {AccountConstants.ACCOUNT_ENTRY_TYPE_BUSINESS},
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+		if (userAccountEntries.isEmpty()) {
+			return null;
 		}
 
-		long userId = _portal.getUserId(httpServletRequest);
+		return userAccountEntries.get(0);
+	}
+
+	private AccountEntry _getPersonAccountEntry(
+			long userId, HttpServletRequest httpServletRequest)
```

Comment:
probably these should be sorted

Reply:
private methods should be already sorted
```
_getAccountEntry
_getBusinessAccountEntry
_getPersonAccountEntry
```

Reply:
Resorted params. Thx 👍🏼